cloud/services/service/load_balancer.go

```diff
@@ -503,3 +504,60 @@ func (s *Service) CheckLoadBalancerDeregisterVm(clockInsideLoop time.Duration, c
 	}
 	return nil
 }
+
+// CheckLoadBalancerRegisterVm ensures that all expected VMs are registered with the load balancer.
+func (s *Service) CheckLoadBalancerRegisterVm(clockInsideLoop time.Duration, clockLoop time.Duration, expectedVmIds []string, spec *infrastructurev1beta1.OscLoadBalancer) error {
+	loadBalancerName, err := s.GetName(spec)
+	if err != nil {
+		return err
+	}
+	clock_time := clock.New()
+	currentTimeout := clock_time.Now().Add(time.Second * clockLoop)
```

Comment:
Why are you adding loops with timeouts in the code here? This is not ideal at all as you are not idempotent when you reach the timeout and thus reconciliation fails.
You should just check once, and if it does not work stop reconciling and requeue a new reconciliation attemp.
Like I did in my 2 PR's improving the security group and vm reconciliations.

Reply:
Thanks for your review,  good point
I am working on your fixes https://github.com/outscale/cluster-api-provider-outscale/pull/385, I am trying to fix the CI to be able to merge it 
And then I will update this